producers/http/handlers.go

```diff
@@ -24,34 +24,47 @@ import (
 	"github.com/gorilla/mux"
 )
 
-// /api/v0/agent
+// /api/v0/node
 func nodeHandler(p *producerImpl) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		var am []interface{}
-		if nodeMetrics, err := p.store.GetByRegex(producers.NodeMetricPrefix + ".*"); err != nil {
-			w.WriteHeader(http.StatusBadRequest)
-		} else {
-			for _, v := range nodeMetrics {
-				am = append(am, v)
-			}
+		nodeMetrics, err := p.store.GetByRegex(producers.NodeMetricPrefix + ".*")
+		if err != nil {
+			httpLog.Error("/api/v0/node - %s", err.Error())
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+		for _, v := range nodeMetrics {
+			am = append(am, v)
+		}
+
+		if len(am) != 0 {
 			encode(am[0], w)
+			return
 		}
+
+		httpLog.Error("/api/v0/node - no content in store.")
+		http.Error(w, "No values found in store", http.StatusBadRequest)
+		return
 	}
 }
 
 // /api/v0/containers
 func containersHandler(p *producerImpl) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		cm := []string{}
-		if containerMetrics, err := p.store.GetByRegex(producers.ContainerMetricPrefix + ".*"); err != nil {
-			w.WriteHeader(http.StatusBadRequest)
-		} else {
-
-			for _, c := range containerMetrics {
-				cm = append(cm, c.(producers.MetricsMessage).Dimensions.ContainerID)
-			}
-			encode(cm, w)
+		containerMetrics, err := p.store.GetByRegex(producers.ContainerMetricPrefix + ".*")
+		if err != nil {
+			httpLog.Error("/api/v0/containers - %s", err.Error())
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		for _, c := range containerMetrics {
+			cm = append(cm, c.(producers.MetricsMessage).Dimensions.ContainerID)
```

Comment:
i'd be tempted to type assert on a new line
```go
a, ok := c.(producers.MetricsMessage)
if !ok {
    return ...
}
```

this will ensure we don't panic if unable to type assert